coresimd/ppsv/api/rotates.rs

```diff
@@ -0,0 +1,104 @@
+//! Implements integer rotates.
+#![allow(unused)]
+
+/// Trait used for overloading rotates
+pub trait Rotate<T> {
```

Comment:
Given that you provide a specific implementation for each vector type below, this doesn't really need to be a trait.

In general we try to keep the traits we use to a minimum, and most of them are sealed (not exposed in the public API of the library).